src/Elders.Cronus/Hosting/CronusBooter.cs

```diff
@@ -8,6 +8,7 @@ public class CronusBooter
     {
         public static void BootstrapCronus(IServiceProvider serviceProvider)
         {
+            CronusLogger.Bootstrap(serviceProvider);
```

Comment:
@mynkow Because of the static logging, the `LoggerFactory` has to be configured as soon as possible. Is that the right place to do that?

Reply:
@ethno2405 this is executed after the `IServiceCollection` initialization. This means that we will not be able to log anything which is happening before the bootstrap. I think a better place will be the first like in [AddCronus()](https://github.com/Elders/Cronus/blob/release-6.0.0/src/Elders.Cronus/Hosting/CronusServiceCollectionExtensions.cs#L29)

Reply:
@mynkow Creating instances of `ILoggerFactory` requires an `IServiceProvider`. `AddCronus()` does not provide that, which means the order of service registrations will matter.
This won't work:
```csharp
IServiceCollection services = new ServiceCollection();
services
    .AddCronus(configuration)
    .AddLogging(x => x.AddConsole().SetMinimumLevel(LogLevel.Trace));
```
and this will:
```csharp
IServiceCollection services = new ServiceCollection();
services
    .AddLogging(x => x.AddConsole().SetMinimumLevel(LogLevel.Trace))
    .AddCronus(configuration);
```

A workaround for this is to use a separate instance of `ServiceCollection` for logging
```csharp
class Program
{
    static void Main(string[] args)
    {
            IServiceCollection loggingServices = new ServiceCollection();
            loggingServices.AddLogging(x => x.AddConsole().SetMinimumLevel(LogLevel.Trace));
            var logger = loggingServices.BuildServiceProvider().GetRequiredService<ILoggerFactory>().CreateLogger<Program>();

            try
            {
                var configuration = new ConfigurationBuilder()
                    .AddJsonFile("appsettings.json")
                    .Build();

                IServiceCollection services = new ServiceCollection();
                services
                    .AddCronus(configuration)
                    .AddLogging(x => x.AddConfiguration(configuration));
            }
            catch (Exception ex)
            {
                logger.LogError(ex, "Error");
            }
    }
}
```

Reply:
@ethno2405 yeah, this is general issue with no clear answer yet. I also think that separate instance of the IServiceCollection might work. Do you see any cons?

Reply:
@mynkow You will have to hardcode the logger providers if you want to log errors from the `ConfigurationBuilder`. If that's not important then you could do something like this:
```csharp
class Program
{
    static void Main(string[] args)
    {
            var configuration = new ConfigurationBuilder()
                    .AddJsonFile("appsettings.json")
                    .Build();
                
            IServiceCollection loggingServices = new ServiceCollection();
            loggingServices.AddLogging(x => x.AddConfiguration(configuration));
            var logger = loggingServices.BuildServiceProvider().GetRequiredService<ILoggerFactory>().CreateLogger<Program>();

            try
            {
                IServiceCollection services = new ServiceCollection();
                services
                    .AddCronus(configuration)
                    .AddLogging(x => x.AddConfiguration(configuration));
...
```